operator/cmd/mesh/manifest-generate_test.go

```diff
@@ -256,6 +257,54 @@ func TestBareSpec(t *testing.T) {
 	})
 }
 
+func TestLocalFilesystem(t *testing.T) {
+	testDataDir = filepath.Join(repoRootDir, "cmd/mesh/testdata/manifest-generate")
+	releaseDir, err := ioutil.TempDir(os.TempDir(), "istio-test-release-*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(releaseDir)
+	cmd := exec.Command("../../release/create_release_charts.sh", "-o", releaseDir)
+	if stdo, err := cmd.Output(); err != nil {
+		t.Fatalf("%s: \n%s", err, string(stdo))
+	}
+	runTestGroup(t, testGroup{
+		{
+			// Use some arbitrary small test input (pilot only) since we are testing the local filesystem code here, not
+			// manifest generation.
+			desc:       "pilot_default",
+			diffSelect: "Deployment:*:istiod",
+			flags:      "--set installPackagePath=" + releaseDir,
+		},
+		{
+			// Specify both charts and profile from local filesystem.
+			desc:       "pilot_default",
+			diffSelect: "Deployment:*:istiod",
+			flags:      fmt.Sprintf("--set installPackagePath=%s --set profile=%s/profiles/default.yaml", releaseDir, releaseDir),
+		},
+	})
+
+}
+
+// TestGithub does a live fetch from github. Alternative is the use a local server but this is would not be an e2e test.
+func TestGithub(t *testing.T) {
+	testDataDir = filepath.Join(repoRootDir, "cmd/mesh/testdata/manifest-generate")
+	runTestGroup(t, testGroup{
+		{
+			// --force is needed for version mismatch.
+			desc:       "github_1.5",
+			diffSelect: "Deployment:*:istiod",
+			flags:      "--force",
+		},
+		{
+			// Same path, using --set.
+			desc:       "github_1.5",
+			diffSelect: "Deployment:*:istiod",
+			flags:      "--force --set installPackagePath=https://github.com/istio/istio/releases/download/1.5.0/istio-1.5.0-linux.tar.gz",
```

Comment:
I don't really think we should have unit tests pulling from github. Or integration tests, if it can be avoided

Reply:
I also have an instinctive aversion to doing this but the downside is a fully hermetic test is quite contrived (i.e. it ends up testing that the internal server & package are correctly set up in the test). The downsides to pulling from an URL (flakiness, brittle tests) are not particularly severe in this case, is there any other downside I'm missing?

Reply:
what is this actually testing though? Let's assume the flag is broken and ignored by istioctl. Won't there still be an istiod deployment?

Reply:
It's testing an installation path for the charts - both local filesystem and github. We can set installPackagePath=/tmp/istio-1.5.0 or https://github.com/istio/istio/releases/download/1.5.0/istio-1.5.0-linux.tar.gz.
My point about the github test is that to set up a server and create a fake release tar to serve from it leaves plenty of room to mess up the test and have the test be passing while the real thing is failing.  Anyway, I'm happy to go with the consensus on this one because I also feel uneasy about it on principle.

Reply:
My question was more about the `diffSelect` - my point was that if we're testing against 1.5.0 charts we should at least test on something that is included in those charts, but not in `master` anymore, so we know it actually pulled them down

Reply:
Yes, if we pull externally I agree there would be value, but with the latest change we are just testing our fake test build so I'm not sure it's worth checking more than basic function.